From 9bc39208067e87ceb1d1fd85e7d1c03786804997 Mon Sep 17 00:00:00 2001
From: Lokesh Vutla <lokeshvutla@ti.com>
Date: Thu, 3 Sep 2015 12:22:16 +0530
Subject: [PATCH 549/800] rtc: omap: Remove enabling of clock source from
 driver-data

External clock source information is passed from dt.
So remove the code that adds this information from driver-data.

Signed-off-by: Lokesh Vutla <lokeshvutla@ti.com>
[zou: Original patch taken from PROCESSOR-SDK-LINUX-AM335X 02_00_01_07]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/rtc/rtc-omap.c |    9 ---------
 1 files changed, 0 insertions(+), 9 deletions(-)

diff --git a/drivers/rtc/rtc-omap.c b/drivers/rtc/rtc-omap.c
index b65078c..e2d7b60 100644
--- a/drivers/rtc/rtc-omap.c
+++ b/drivers/rtc/rtc-omap.c
@@ -131,7 +131,6 @@
 struct omap_rtc;
 
 struct omap_rtc_device_type {
-	bool has_32kclk_en;
 	bool has_irqwakeen;
 	bool has_pmic_mode;
 	bool has_power_up_reset;
@@ -566,7 +565,6 @@ static const struct omap_rtc_device_type omap_rtc_default_type = {
 };
 
 static const struct omap_rtc_device_type omap_rtc_am3352_type = {
-	.has_32kclk_en	= true,
 	.has_irqwakeen	= true,
 	.has_pmic_mode	= true,
 	.lock		= am3352_rtc_lock,
@@ -661,13 +659,6 @@ static int omap_rtc_probe(struct platform_device *pdev)
 	 */
 	rtc_writel(rtc, OMAP_RTC_INTERRUPTS_REG, 0);
 
-	/* enable RTC functional clock */
-	if (rtc->type->has_32kclk_en) {
-		reg = rtc_read(rtc, OMAP_RTC_OSC_REG);
-		rtc_writel(rtc, OMAP_RTC_OSC_REG,
-				reg | OMAP_RTC_OSC_32KCLK_EN);
-	}
-
 	/* clear old status */
 	reg = rtc_read(rtc, OMAP_RTC_STATUS_REG);
 
-- 
1.7.5.4

